test/e2e_env/kubernetes/gateway/delegated/meshtimeout.go

```diff
@@ -0,0 +1,122 @@
+package delegated
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	meshretry_api "github.com/kumahq/kuma/pkg/plugins/policies/meshretry/api/v1alpha1"
+	"github.com/kumahq/kuma/pkg/plugins/policies/meshtimeout/api/v1alpha1"
+	"github.com/kumahq/kuma/test/framework"
+	"github.com/kumahq/kuma/test/framework/client"
+	"github.com/kumahq/kuma/test/framework/envs/kubernetes"
+)
+
+func MeshTimeout(config *Config) func() {
+	GinkgoHelper()
+
+	return func() {
+		BeforeAll(func() {
+			Expect(framework.DeleteMeshResources(
+				kubernetes.Cluster,
+				config.Mesh,
+				v1alpha1.MeshTimeoutResourceTypeDescriptor,
+			)).To(Succeed())
+
+			Expect(framework.DeleteMeshResources(
+				kubernetes.Cluster,
+				config.Mesh,
+				meshretry_api.MeshRetryResourceTypeDescriptor,
+			)).To(Succeed())
+		})
+
+		framework.E2EAfterEach(func() {
```

Comment:
Do we need `framework.E2EAfterEach` lets add import `. "github.com/kumahq/kuma/test/framework"` and use `E2EAfterEach`

Reply:
We can't as there is conflict with `Config` structs